itest/rust/src/object_tests/init_level_test.rs

```diff
@@ -120,3 +122,40 @@ fn on_init_scene() {
 pub fn on_init_editor() {
     // Nothing yet.
 }
+
+#[derive(GodotClass)]
+#[class(base=Object, init)]
+struct MainLoopCallbackSingleton {
+    #[init(val=RenderingServer::singleton())]
+    rs: Gd<RenderingServer>,
+    #[init(val=RenderingServer::singleton().texture_2d_placeholder_create())]
+    tex: Rid,
+}
```

Comment:
`#[init]` with very complex expressions is imo an antipattern, here it would be better to define a `init()` function. Integration tests are also _much_ easier to refactor if not everything depends on macro magic.

But also, why does the singleton `rs` need to be cached at all? Can we not keep the itest focused on the functionality being tested, and thus as simple as possible? 🙂